internal/cluster/cluster.go

```diff
@@ -952,9 +952,52 @@ func (m *Manager) SetVipsData(ctx context.Context, c *common.Cluster, apiVip, in
 	return nil
 }
 
+func (m *Manager) uploadClusterDataAsFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	log := logutil.FromContext(ctx, m.log)
+	events, err := m.eventsHandler.V2GetEvents(ctx, c.ID, nil, nil)
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload events file")
+		return err
+	}
+
+	eventsAsData, err := json.MarshalIndent(events, "", " ")
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload events file")
+		return err
+	}
+	fileName := fmt.Sprintf("%s/logs/cluster/events.json", c.ID)
+
+	err = objectHandler.Upload(ctx, eventsAsData, fileName)
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload events file")
+		return err
+	}
+
+	clusterData, err := json.MarshalIndent(c, "", " ")
```

Comment:
Let's remove pull secret before the upload
Not sure if we have other sensitive info

Reply:
done